assets/lumi/shaders/common/shadow.glsl

```diff
@@ -70,7 +70,7 @@ float calcShadowFactor(in sampler2DArrayShadow shadowMap, vec4 shadowViewPos) {
         float shadowFactor = sampleShadowPCF(shadowMap, shadowCoords.xyz, float(cascade));
     #endif
     
-    return shadowFactor;
+    return min(1, shadowFactor + frx_rainGradient() * 0.8 + frx_thunderGradient() * 0.2);
```

Comment:
The global shadow library is for sampling the shadow map according to the user configuration (pcf, etc). It's not for atmosphere calculation which needs to be isolated in the atmosphere library. With this code sunlight would randomly reach underground during rain.